packages/quantic/force-app/main/default/lwc/quanticRecentQueries/quanticRecentQueries.js

```diff
@@ -0,0 +1,83 @@
+import { LightningElement, api, track } from 'lwc';
+import { registerComponentForInit, initializeWithHeadless } from 'c/quanticHeadlessLoader';
+import { getItemFromLocalStorage, setItemInLocalStorage } from 'c/quanticUtils';
+import recentQueriesLabel from '@salesforce/label/c.quantic_RecentQueries';
+import emptyListLabel from '@salesforce/label/c.quantic_EmptyRecentQueriesLabel';
+
+export default class QuanticRecentQueries extends LightningElement {
+  labels = {
+    recentQueriesLabel,
+    emptyListLabel,
+  }
+  
+  /** @type {import("coveo").RecentQueriesState} */
+  @track state;
+  /** @type {string} */
+  @api label = this.labels.recentQueriesLabel;
+  /** @type {string} */
+  @api engineId;
+  /** @type {import("coveo").RecentQueriesList} */
+  recentQueriesList;
+  /** @type {()=> void} */
+  unsubscribe;
+  /** @type {boolean} */
+  isCollapsed = false;
+  /** @type {string} */
+  collapseIcon = 'utility:dash';
+  /** @type {number} */
+  @api maxLength = 10;
```

Comment:
Considering this component is customer facing should we add jsdoc documentation on each exposed property ?

Reply:
Definitely! but, given the jsdoc setup isn't implemented yet, we'll likely have to modify whatever we put here when we add proper jsdoc comments for all the other components. So with that in mind I'd say it's probably not critical that more details be added in this PR right now. My 2 cents